nodeJS/express/views.md

```diff
@@ -0,0 +1,208 @@
+### Introduction
+
+Views are the user-facing part of the application, in this case, HTML files. We've dealt with views in an earlier project where the server would send HTML files to the user.  These files are static, but many of our use cases require views to be dynamic w.r.t. data.
+
+Hence, we use template engines to create our views. As the name suggests, we write template files in our codebase that get transformed into HTML when we respond to a server request. Any variables defined in our template files are replaced with actual data. Additionally, we can insert conditional and/or loop logic into our template file, e.g. render the user's username once they have logged in. This would not be possible with plain HTML.
+
+In this course, we will use [EJS](https://ejs.co/). EJS's syntax is very similar to that of HTML, meaning that the learning curve is relatively low compared to other template engines.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- How to setup EJS in an Express project
+- How to use EJS
+
+### Setting up EJS
+
+Let's get started with EJS! Install EJS into your application by typing the following into your terminal:
+
+```bash
+npm install ejs
+```
+
+At the root of your project, create a subfolder called `views`.
+
+Next, we need to let our app know that we intend to use `EJS` as a template engine, as well as where to look for view files.
+
+In your `app.js` file, set the following application properties:
+
+```javascript
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "ejs");
+```
+
+This enables EJS as the view engine, and that our app should look for templates in the `/views` subdirectory.
+
+### EJS syntax
+
+In EJS, the `<%` and `%>` tags allow us to use JavaScript. This lets us write conditional statements, `for` loops, as well as use variables.
+
+In order to output a variable as a value, we use the `<%=` tag.
+
+Here's a quick example that includes arrays and loop logic.
+
+```ejs
+<% const animals = ["Cat", "Dog", "Lemur", "Hawk"] %>
+
+<% animals.map(animal => { %>
+- <%= animal%>s are cute <% }) %>
```

Comment:
```suggestion
<% const animals = ["Cat", "Dog", "Lemur", "Hawk"] %>

<ul>
  <% animals.map((animal) => { %>
    <li><%= animal %>s are cute</li>
  <% }) %>
</ul>
```
Suggestion - just to put loops into context (ul/li).
I think we should at the very least format the JS parts on different lines e.g.

```ejs
<% animals.map((animal) => { %>
  <%= animal %>s are cute
<% }) %>
```